### PR TITLE
Replace DWindow to DMainWindow

### PR DESCRIPTION
**Pull request by rekols**

### CODE REVIEW

**Comment by sbwtw** [comment]
Thanks for your contribute!
We have a code-review specification, please submit your commits to this site: https://cr.deepin.io/#/admin/projects/deepin-deb-installer

File: debinstaller.cpp [hunk 1]
```diff
@@ -88,7 +93,7 @@ void DebInstaller::onPackagesSelected(const QStringList &packages)
         m_centralLayout->addWidget(singlePage);
     } else {
         // multiple packages install
-        setTitle(tr("Bulk Install"));
+        //setTitle(tr("Bulk Install"));
```
Comment (sbwtw) [inline debinstaller.cpp]: please revert this

Comment (rekols) [inline debinstaller.cpp]: setWindowTitle?

File: debinstaller.h [hunk 2]
```diff
@@ -29,6 +29,7 @@ private slots:
     SingleInstallPage *backToSinglePage();
 
 private:
+    QWidget *m_widget;
```
Comment (sbwtw) [inline debinstaller.h]: don't know why you need this additional widget

File: debinstaller.cpp [hunk 3]
```diff
@@ -28,14 +30,17 @@ DebInstaller::DebInstaller(QWidget *parent)
...
     m_centralLayout->setSpacing(0);
 
-    setLayout(m_centralLayout);
     setFixedSize(480, 380);
+    this->titleBar()->setWindowFlags(Qt::WindowTitleHint | Qt::WindowMinimizeButtonHint | Qt::WindowSystemMenuHint | Qt::WindowCloseButtonHint);
     setWindowTitle(tr("Deepin Package Manager"));
     setWindowIcon(QIcon::fromTheme("deepin-deb-installer"));
-    setTitleIcon(QIcon::fromTheme("deepin-deb-installer").pixmap(24, 24));
-    setTitle(QString());
+    setCentralWidget(m_widget);
```
Comment (sbwtw) [inline debinstaller.cpp]: why don't you set a layout to DMainWindow directly?

File: debinstaller.cpp [hunk 4]
```diff
@@ -28,14 +30,17 @@ DebInstaller::DebInstaller(QWidget *parent)
     m_centralLayout->setContentsMargins(0, 0, 0, 0);
     m_centralLayout->setSpacing(0);
 
-    setLayout(m_centralLayout);
     setFixedSize(480, 380);
+    this->titleBar()->setWindowFlags(Qt::WindowTitleHint | Qt::WindowMinimizeButtonHint | Qt::WindowSystemMenuHint | Qt::WindowCloseButtonHint);
```
Comment (sbwtw) [inline debinstaller.cpp]: unuseful `this->`

File: debinstaller.h (hunk 2 above)
Comment (rekols) [inline debinstaller.h]: because it will show two window

File: debinstaller.cpp (hunk 4 above)
Comment (rekols) [inline debinstaller.cpp]: ok

**Comment by sbwtw** [comment]
hey, please follow this guide https://github.com/linuxdeepin/developer-center/wiki/Contribution-Guidelines-for-Developers to know how we receive contributions, have fun ;)

I'll close this issue later.

**Pull request closed without merging**